docs/planning/adr/adr-002-framework-agnostic-design.md

```
@@ -0,0 +1,171 @@
+
+### Implementation
+
+#### Components Affected
+
+1. **cloudflare-auth/core/**: Framework-agnostic JWT validation, models
+2. **cloudflare-auth/fastapi/**: Middleware, dependencies, request handling
+3. **pyproject.toml**: Optional dependencies for framework extras
+
+#### Target Structure
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)